doc/user/content/manage/access-control/_index.md

```diff
@@ -10,116 +10,298 @@ menu:
+GRANT USAGE ON SCHEMA d1.s1 TO "user1@company.com", "user2@company.com";
+GRANT USAGE ON CLUSTER c1 to "user1@company.com", "user2@company.com";
+
+-- Example: grant individual roles permission to create clusters and databases
+GRANT CREATECLUSTER, CREATEDB on system to "user1@company.com";
+```
+
+This strategy allows you to skip setting up a complex hierarchy of roles
+upfront, at the cost of manual work. As you might expect, this strategy does
+not scale, so you might want to rethink your RBAC strategy if your user base
+grows past a reasonable number (say, 10+).
+
+#### Functional roles
+
+If your Materialize user base is made up of business functions that clearly map
+to distinct access profiles, a common strategy is to map those functions to
+roles in your RBAC hierarchy. You can then grant and revoke privileges to/from
+functional roles, and any user that is a member of that role will inherit
+them.
+
+As an example, Data Engineers might need a larger scope of permissions to create
+and evolve the data model, while Data Analysts might only need read permissions
+to query Materialize using BI tools.
+
+```sql
+-- Use SHOW ROLES to list existing roles in the system, which are 1:1 with invited users
+SHOW ROLES;
+
+-- Step 1: create the data_analyst role
+CREATE ROLE data_analyst;
+
+-- Step 2: grant usage on the database & schema to the data_analyst role
+GRANT USAGE ON DATABASE d1 TO data_analyst;
+GRANT USAGE ON SCHEMA d1.s1 TO data_analyst;
+GRANT USAGE ON CLUSTER c1 to data_analyst;
+
+-- Step 3: grant select to all objects in the d1.s1 schema to the data_analyst role.
+-- For pre-existing objects in the schema (skip if there are none!)
+GRANT SELECT ON ALL TABLES IN SCHEMA d1.s1 TO data_analyst;
+-- For future objects created in the schema
+ALTER DEFAULT PRIVILEGES FOR ALL ROLES IN SCHEMA d1.s1 GRANT SELECT ON TABLES TO data_analyst;
+
+-- Step 4: create the data_engineer role
+CREATE ROLE data_engineer;
+
+-- Step 5: grant usage & create on the database & schema to the data_engineer role
+GRANT USAGE, CREATE ON DATABASE d1 TO data_engineer;
+GRANT USAGE, CREATE ON SCHEMA d1.s1 TO data_engineer;
+GRANT USAGE, CREATE ON CLUSTER c1 to data_engineer;
+
+-- Step 6: grant all privileges to all objects in the d1.s1 schema to the data_engineer role
+-- For pre-existing objects in the schema (skip if there are none!)
+GRANT ALL PRIVILEGES ON ALL TABLES IN SCHEMA d1.s1 TO read_write;
+-- For future objects created in the schema
+ALTER DEFAULT PRIVILEGES FOR ALL ROLES IN SCHEMA d1.s1 GRANT ALL PRIVILEGES ON TABLES TO read_write;
+
+-- Step 7: add member(s) to the respective role
+GRANT data_analyst TO "user1@company.com", "user2@company.com";
+GRANT data_engineer TO "user3@company.com";
+```
+
+This strategy allows you to more efficiently manage permissions for multiple
+users at a time, as well as create an intuitive hierarchy based on each
+functional role's expected usage of Materialize.
+
+#### Environment roles
```

Comment:
Why include this section? Just to get people to let us know if they want us to implement this?

Reply:
I'm partial to this. @chaas left a comment in the original RBAC Notion doc about including it to assess user interest, so...it might, worst case, help accelerate prioritization?

Reply:
Yeah we're waiting on customer ask(s) to put in the time since it's nontrivial effort, but it is doable and we can prioritize if customers need it. I like the chance to get direct signal on priority this way.